src/datasets/iterable_dataset.py

```diff
@@ -349,20 +352,23 @@ def __init__(
     def _head(self, n=5):
         return _examples_to_batch([x for key, x in islice(self._iter(), n)])
 
-    @property
-    def _effective_seed(self):
-        if self._shuffling:
-            return self._shuffling.seed + self._epoch if self._shuffling.seed is not None else None
+    def _effective_generator(self):
+        if self._shuffling and self._epoch == 0:
+            return self._shuffling.generator
+        elif self._shuffling:
+            # Create effective seed using self._epoch (we substract in order to avoir overflow in long_scalars)
+            effective_seed = deepcopy(self._shuffling.generator).integers(0, 1 << 63) - self._epoch
```

Comment:
If I'm not mistaken, `effective_seed` can be a negative integer after this line, which will result in an error in `np.random.default_rng(effective_seed)`.
```suggestion
            effective_seed = deepcopy(self._shuffling.generator).integers(0, 1 << 63) - self._epoch
            effective_seed = 1 << 63 + effective seed if effective_seed < 0 else effective_seed
```

Reply:
Good catch ! thanks